File: src/main/java/ru/menshevva/demoapp/ui/admin/roles/RoleEditView.java
package ru.menshevva.demoapp.ui.admin.roles;

import com.vaadin.flow.component.Tag;
import com.vaadin.flow.component.dependency.JsModule;
import com.vaadin.flow.component.littemplate.LitTemplate;
import com.vaadin.flow.component.template.Id;
import com.vaadin.flow.component.textfield.TextField;
import com.vaadin.flow.data.binder.Binder;
import com.vaadin.flow.data.binder.ValidationException;
import ru.menshevva.demoapp.dto.role.RoleData;


@Tag("role-edit-view")
@JsModule("./src/admin/roles/role-edit-view.ts")
public class RoleEditView extends LitTemplate {

    @Id
    private TextField roleName;

    @Id
    private TextField roleDescription;

    private final Binder<RoleData> binder = new Binder<>();

    public RoleEditView() {
        binder.forField(roleName)
                .asRequired("Имя роли обязательно")
                .bind(RoleData::getRoleName, RoleData::setRoleName);
        binder.forField(roleDescription)
                .asRequired("Описание роли обязательно")
                .bind(RoleData::getRoleDescription, RoleData::setRoleDescription);
    }

    public void setRoleData(RoleData roleData) {
        binder.readBean(roleData);
    }

    public void getRoleData(RoleData roleData) throws ValidationException {
        binder.writeBean(roleData);
    }

}

File: src/main/frontend/src/admin/roles/role-edit-view.ts
import {html, LitElement} from "lit";
import {customElement} from "lit/decorators.js";
import '@vaadin/text-field';

@customElement("role-edit-view")
class RoleEditView extends LitElement {

    render() {
        return html `
            <div>
                <vaadin-text-field id="roleName" label="Наименование"></vaadin-text-field></vaadin-text-field>
                <vaadin-text-field id="roleDescription" label="Описание"></vaadin-text-field></vaadin-text-field>
            </div>
        `;
    }

}